File: frontend/src/Domain/Dtos/UserSubjects.ts
import FinalGradeDto from '@/domain/dtos/FinalGradeDto.ts';
import GradeDto from '@/domain/dtos/GradeDto.ts';
import PresenceDto from '@/domain/dtos/PresenceDto.ts';
import ClassSubjectSemesterDto from '@/domain/dtos/ClassSubjectSemesterDto.ts';

interface UserSubjects {
  classSubjectSemesters: ClassSubjectSemesterDto[];
  finalGrades: FinalGradeDto[];
  grades: GradeDto[];
  presences: PresenceDto[];
}

export default UserSubjects;

File: Application/Dto/School/UserSubjects.cs
using Domain.Entities_v2.School;

namespace Application.Dto.School;

public class UserSubjects
{
    public ClassSubjectSemesterDto[] ClassSubjectSemesters { get; set; } = null!;
    public FinalGradeDto[] FinalGrades { get; set; } = null!;
    public GradeDto[] Grades { get; set; } = null!;
    public PresenceDto[] Presences { get; set; } = null!;
}